docs/ioc/user-guides/reccaster.rst

```diff
@@ -0,0 +1,102 @@
+RecCaster configuration
+=======================
+
+RecCaster is an EPICS support module
+that sends PV names and IOC metadata to a "RecCeiver" server.
+This setup is often used to send such metadata to a ChannelFinder database.
+
+.. seealso::
+
+   If you want to set up a RecCeiver and ChannelFinder server,
+   examine the guide
+   :doc:`../../nixos-services/user-guides/channel-finder`.
+
+Configuration
+-------------
+
+To configure RecCaster,
+first add it to your build environment:
+
+.. code-block:: diff
+   :caption: :file:`flake.nix`
+
+            # Add your support modules here:
+            # ---
+   -        #support.modules = with pkgs.epnix.support; [ StreamDevice mySupportModule ];
+   +        support.modules = with pkgs.epnix.support; [ reccaster ];
+
+Make sure your app depends on the RecCaster library and DBD file:
+
+.. code-block:: make
+   :caption: :file:`{example}App/src/Makefile`
+
+   # Replace "example" by the name of your application
+   example_DBD += reccaster.dbd
+   example_LIBS += reccaster
+
+And load the RecCaster database:
+
+.. code-block:: csh
+   :caption: :file:`iocBoot/ioc{Example}/st.cmd`
+
+   ## Load RecCaster records
+   ## Make sure to choose your prefix by setting the 'P' macro
+   dbLoadRecords("${RECCASTER}/db/reccaster.db", "P=YOUR-PV-PREFIX:")
+
+This configuration is enough to start the RecCaster client.
+
+.. note::
+
+   RecCaster doesn't need the address of the RecCeiver server.
+   The RecCeiver server scans for available IOCs on the network.
+
+Firewall
+--------
+
+To accept connection from the RecCeiver service,
+the firewall needs to allow the UDP announcert port,
+which by default is 5049.
+
+If your IOC is a NixOS machine,
+you can open the firewall with this NixOS configuration:
+
+.. code-block:: nix
+   :caption: Opening the firewall for RecCeiver
+
+   networking.firewall.allowedUDPPorts = [5049];
+
+Checking synchronization status
+-------------------------------
+
+To check the RecCaster status of your IOC,
+you can run:
+
+.. code-block:: bash
+
+   # Replace 'YOUR-PV-PREFIX:' with your PV prefix
+   caget "YOUR-PV-PREFIX:Msg-I"
+   caget "YOUR-PV-PREFIX:State-Sts"
+
+Adding IOC metadata
+-------------------
+
+You can choose to send more information to the RecCeiver server.
+
+If you want to add metadata about the global IOC,
+set it as environment variable
+and expose it using the ``addReccasterEnvVars`` function.
+For example:
+
+.. code-block:: csh
+   :caption: :file:`iocBoot/ioc{Example}/st.cmd`
+
+   epicsEnvSet("CONTACT", "mycontact")
+   addReccasterEnvVars("CONTACT")
+
```

Comment:
some exist by default and are used by Phoebus: 
epicsEnvSet("IOCNAME", "abc-ioc1")
epicsEnvSet("ENGINEER", "Fred")
epicsEnvSet("LOCATION", "Rack 123")

It is mentionned in the example after but should be mentionned here also

Reply:
Used by Phoebus, as in their are displayed specially?

Reply:
yes
![image-20230616152439-1](https://github.com/user-attachments/assets/e545a845-921b-494f-8080-60c935cc227f)

see here the columns ENGINEER, IOCNAME. Location is probably hidden in the right

Reply:
I can't reproduce with those variables. With an empty `environment_vars` in the RecCeiver config, I get:

![image](https://github.com/user-attachments/assets/92aa2639-4f36-4982-a21d-dfe7701b42b3)

I think I'll document `hostName`, `iocName`, and `iocid` then

Reply:
i don't understand what I am looking at

Reply:
Ah, I see I think the issue, I'll update the doc.

Reply:
Done